### PR TITLE
Add support for interrupting execution with ^C

### PR DESCRIPTION
**Pull request by Axel-Naumann**
https://github.com/root-project/cling/pull/245

### CODE REVIEW

File: interpreter/cling/lib/Utils/Signal.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+//------------------------------------------------------------------------------
+// CLING - the C++ LLVM-based InterpreterG :)
```
Comment (amadio) [inline interpreter/cling/lib/Utils/Signal.cpp]: Why `InterpreterG`?

Comment (Axel-Naumann) [reply interpreter/cling/lib/Utils/Signal.cpp]: An old bad joke to make `cling` an abbreviation ;-)

Comment (amadio) [reply interpreter/cling/lib/Utils/Signal.cpp]: Nevermind. I thought the author was Axel and that he asked me for a review.
Anyway, the `G` in the end is not a typo, it turns out :smiley:

**Comment by Axel-Naumann** [comment]
@phsft-bot build!

**Comment by Axel-Naumann** [comment]
@phsft-bot build!

**Comment by amadio** [comment]
I guess this can be closed? See [ROOT-9906](https://sft.its.cern.ch/jira/browse/ROOT-9906).

**Comment by Axel-Naumann** [comment]
Addressed independently.

**Pull request closed without merging**